File: src/main/java/common/Shape.java
package common;

public class Shape {

    public static final String HEXAGON = "HEXAGON";
    public static final String OCTAGON = "OCTAGON";
    public static final String RECTANGLE = "RECTANGLE";
    public static final String TRIANGLE = "TRIANGLE";
    public static final String DIAMOND = "DIAMOND";
    public static final String PENTAGON = "PENTAGON";
    public static final String STAR = "STAR";
    public static final String BALL = "BALL";
    public static final String NO_SHAPE = "NO_SHAPE";


    public static String getColor(String shape) {
        if(shape.endsWith("<>")) {
            return shape.replace("<>", "").trim();
        } else {
            int hyphen = shape.indexOf("-");
            if(hyphen > 0) {
                return shape.substring(0, hyphen);
            } else {
                return shape;
            }
        }
    }

    public static String getSuffix(String shape) {
        if (HEXAGON.equals(shape)) {
            return "-H";
        } else if (OCTAGON.equals(shape)) {
            return "-O";
        } else if (RECTANGLE.equals(shape)) {
            return "-R";
        } else if (TRIANGLE.equals(shape)) {
            return "-T";
        } else if (DIAMOND.equals(shape)) {
            return "<>";
        } else if (PENTAGON.equals(shape)) {
            return "-P";
        } else if (STAR.equals(shape)) {
            return "-S";
        } else {
            return "";
        }
    }

    public static String getShape(String obj) {
        if (obj.isEmpty()) {
            return NO_SHAPE;
        } else if (obj.endsWith("-H")) {
            return HEXAGON;
        } else if (obj.endsWith("-O")) {
            return OCTAGON;
        } else if (obj.endsWith("-R")) {
            return RECTANGLE;
        } else if (obj.endsWith("-T")) {
            return TRIANGLE;
        } else if (obj.endsWith("<>")) {
            return DIAMOND;
        } else if (obj.endsWith("-P")) {
            return PENTAGON;
        } else if (obj.endsWith("-S")) {
            return STAR;
        } else {
            return "BALL";
        }
    }

    public static String getString(String color, String shape) {
        return color + getSuffix(shape);
    }

}

File: src/main/kotlin/common/Shape_.kt
package common

class Shape_ {

    companion object {

        val HEXAGON = "HEXAGON"
        val OCTAGON = "OCTAGON"
        val RECTANGLE = "RECTANGLE"
        val TRIANGLE = "TRIANGLE"
        val DIAMOND = "DIAMOND"
        val PENTAGON = "PENTAGON"
        val STAR = "STAR"
        val BALL = "BALL"
        val NO_SHAPE = "NO_SHAPE"


        fun getColor(shape: String): String =
                when (shape.endsWith("<>")) {
                    true -> shape.replace("<>", "").trim()
                    false -> when (shape.indexOf("-") > 0) {
                        true -> shape.substring(0, shape.indexOf("-"))
                        false -> shape
                    }
                }

        fun getSuffix(shape: String): String =
                when (shape) {
                    HEXAGON -> "-H"
                    OCTAGON -> "-O"
                    RECTANGLE -> "-R"
                    TRIANGLE -> "-T"
                    DIAMOND -> "<>"
                    PENTAGON -> "-P"
                    STAR -> "-S"
                    else -> ""
                }

        fun getShape(obj: String): String =
                when {
                    obj.isEmpty() -> NO_SHAPE
                    obj.endsWith("-H") -> HEXAGON
                    obj.endsWith("-O") -> OCTAGON
                    obj.endsWith("-R") -> RECTANGLE
                    obj.endsWith("-T") -> TRIANGLE
                    obj.endsWith("<>") -> DIAMOND
                    obj.endsWith("-P") -> PENTAGON
                    obj.endsWith("-S") -> STAR
                    else -> "BALL"
                }

        fun getString(color: String, shape: String): String =
                color + getSuffix(shape)


    }

}
